cmake/patch_llvm.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python3
+
+"""
+Script to apply a set of patches to llvm-project sources.
+"""
+
+import argparse
+import os
+import subprocess
+import sys
+
+
+def main():
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument(
+        "patchdir",
+        help="Set of patches to apply. This should be a directory containing one or more ordered *.patch files.",
+    )
+    parser.add_argument(
+        "--llvm_dir",
+        help="Directory of the llvm-project git checkout, if not the current directory.",
+    )
+    parser.add_argument(
+        "--method",
+        choices=["am", "apply"],
+        default="apply",
+        help="Git command to use. git am will add each patch as a commit, whereas git apply will leave patched changes staged.",
+    )
+    parser.add_argument(
+        "--reset",
+        help="Clean and reset the repo to a specified commit before patching.",
```

Comment:
The help message says the option should reset the repo to a "specified commit", but the logic for it resets the repo to `HEAD`.

Reply:
Added the missing arg.